doc/langref/test_basic_slices.zig

```diff
@@ -42,6 +42,10 @@ test "basic slices" {
 
     // Note that `slice.ptr` does not invoke safety checking, while `&slice[0]`
     // asserts that the slice has len > 0.
+
+    // Empty slices can be created like this
+    const empty: []u8 = &.{};
```

Comment:
I think it would be nice to teach the more explicit `&[0]u8{}` first and then explain that `&.{}` is a shorthand.

Reply:
I don't think it's particularly helpful to explain the more verbose version. However, I would say the comment here is incomplete -- the interesting thing here is the mutability of the slice:
```diff
-    // Empty slices can be created like this
+    // A zero-length initialization can always be used to create an empty slice, even if the slice is mutable.
+    // This is because the pointed-to data is zero bits long, so its immutability is irrelevant.
```

Reply:
Incorporated both of your comments. @mlugg can you think of a test assertion that would demonstrate your point?